rbac/common/manager/base_relationship.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2018 Contributors to Hyperledger Sawtooth
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+import logging
+
+from rbac.common.sawtooth.client_sync import ClientSync
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseRelationship:
+    def __init__(self):
+        """Objects and methods shared across relationship libraries"""
+        self.client = ClientSync()
+
+    @property
+    def name(self):
+        raise NotImplementedError("Class must implement this property")
+
+    @property
+    def container_proto(self):
+        # return protobuf.role_state_pb2.RoleRelationshipContainer
+        raise NotImplementedError("Class must implement this property")
+
+    def address(self, object_id, target_id):
+        raise NotImplementedError("Class must implement this method")
+
+    def check(self, object_id, target_id):
```

Comment:
The API path results in:
role.member.check(object_id=role_id, target_id=user_id)
role.admin.check(object_id=role_id, target_id=user_id)
role.owner.check(object_id=role_id, target_id=user_id)

Reply:
Understood, but the purpose and behavior of the name is unclear since the example you does not include the output. How about:
if not has_relationship(source, target)